src/main/java/witchinggadgets/client/render/TileRenderIceGen.java

```diff
@@ -0,0 +1,171 @@
+		renderPixelBlock(tes, 0, 0, 0, 1, 0.1875, 1,0,0,1,0.1875);
+		renderPixelBlock(tes, 0.0625, 0.1875, 0.0625, 0.9375, 0.3125, 0.9375,0,0,1,0.125);
+		renderPixelBlock(tes, 0, 0.8125, 0, 1, 0.9375, 1,0,0,1,0.1875);
+
+		ClientUtilities.bindTexture("thaumcraft:textures/blocks/arcane_stone.png");
+		renderPixelBlock(tes, 0, 1, 0, 1, 1, 1,0,0,1,1);
+		renderPixelBlock(tes, 0, 0.9375, 0, 1, 0.999, 1,0,0,0.125,0.125);
+		renderPixelBlock(tes, 0.0625, 0.3125, 0.0625, 0.9375, 0.3125, 0.9375,0.0625,0.0625,0.9375,0.9375);
+		renderPixelBlock(tes, 0, 0.8125, 0, 1, 0.8125, 1,0,0,1,1);
+		ClientUtilities.bindTexture("thaumcraft:textures/blocks/pedestal_top.png");
+		renderPixelBlock(tes, 0, 0.1875, 0, 0.125, 0.8125, 0.125,0,0,0.125,1);
+		renderPixelBlock(tes, 0, 0.1875, 0.875, 0.125, 0.8125, 1,0,0,0.125,1);
+		renderPixelBlock(tes, 0.875, 0.1875, 0, 1, 0.8125, 0.125,0,0,0.125,1);
+		renderPixelBlock(tes, 0.875, 0.1875, 0.875, 1, 0.8125, 1,0,0,0.125,1);
+
+		if(!tile.facing.equals(ForgeDirection.UP) && !tile.facing.equals(ForgeDirection.DOWN))
+			ClientUtilities.bindTexture("thaumcraft:textures/blocks/liftertop.png");
+		else
+			ClientUtilities.bindTexture("thaumcraft:textures/blocks/lifterside.png");
+		renderPixelBlock(tes, 0.125, 0.125, 0.0624, 0.875, 0.875, 0.0625, 0.125,0.125,0.875,0.875);
+		ClientUtilities.bindTexture("thaumcraft:textures/blocks/lifterside.png");
+		renderPixelBlock(tes, 0.125, 0.125, 0.9375, 0.875, 0.875, 0.9376, 0.125,0.125,0.875,0.875);
+		renderPixelBlock(tes, 0.9375, 0.125, 0.125, 0.9376, 0.625, 0.875, 0.125,0.125,0.875,0.625);
+		renderPixelBlock(tes, 0.0624, 0.125, 0.125, 0.0625, 0.625, 0.875, 0.125,0.125,0.875,0.625);
+
+		ClientUtilities.bindTexture("thaumcraft:textures/models/Bore.png");
+		if(tile.facing.equals(ForgeDirection.UP))
+		{
+			renderPixelBlock(tes, 0.375, 1, 0.375, 0.625, 1.125, 0.625, 0.859375,0.65625,0.8984375,0.71875);
+			renderPixelBlock(tes, 0.375, 1.125, 0.375, 0.625, 1.125, 0.625, 0.828125,0.875,0.8671875,0.953125);
+		}
+		else if(tile.facing.equals(ForgeDirection.DOWN))
+		{
+			renderPixelBlock(tes, 0.375,-0.125, 0.375, 0.625, 0, 0.625, 0.859375,0.65625,0.8984375,0.71875);
+			renderPixelBlock(tes, 0.375,-0.125, 0.375, 0.625,-0.125, 0.625, 0.828125,0.875,0.8671875,0.953125);
+		}
+		else
+		{
+			renderPixelBlock(tes, 0.4075, 0.4075, 0, 0.5925, 0.5925, 0.0624, 0.859375,0.65625,0.8984375,0.71875);
+			renderPixelBlock(tes, 0.375, 0.375,-0.125, 0.625, 0.625, 0, 0.859375,0.65625,0.8984375,0.71875);
+			renderPixelBlock(tes, 0.375, 0.375,-0.125, 0.625, 0.625,-0.125, 0.828125,0.875,0.8671875,0.953125);
+		}
+
+		if(tile.getWorldObj()==null || tile.getWorldObj().getBlockPowerInput(tile.xCoord, tile.yCoord, tile.zCoord)<=0 && !tile.getWorldObj().isBlockIndirectlyGettingPowered(tile.xCoord, tile.yCoord, tile.zCoord))
+		{
+			double slowTick = tick/4.0;
+			double loopTick = slowTick*1.65;
+			double inc = 1.0/512.0;
+			GL11.glEnable(3042);
```

Comment:
same here. I don't know surrounding code so maybe no state is actually leaked, just asking to be sure.

Reply:
I was copying what was there but extra protection shouldn’t hurt